android/source/VideoLocker/src/org/edx/mobile/view/CourseLectureListActivity.java

```diff
@@ -191,15 +191,18 @@ private void loadData() {
     @Override
     protected void onStart() {
         super.onStart();
-        adapter.setStore(db, storage, enrollment.getCourse().getId());
-        loadData();
-        handler.sendEmptyMessage(MSG_UPDATE_PROGRESS);
         try{
+            //Check if the adapter and enrollment is not null
+            if(adapter!=null && enrollment!=null){
+                adapter.setStore(db, storage, enrollment.getCourse().getId());
+                loadData();
+            }
+
```

Comment:
@shahidtamboli Adapter shouldn't be null and should be created new if it is null.